### PR TITLE
feat(route/keylol): Update keylol

### PR DESCRIPTION
**Pull request by kennyfong19931**
## Involved Issue / 该 PR 相关 Issue

无

## Example for the Proposed Route(s) / 路由地址示例

```routes
/keylol/f161-1
/keylol/fid=234&typeid=875
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
试用现存路由时，发现每个项目第一张图都是广告，动手弄走把它，顺便加强了项目的读取

- 移除广告
- 找回原作者的使用教学
- 以作者名字取代编号 (如有)
- 支持显示包含目录帖文的全部内容，常用于特卖帖 (如示例中的 [`SEGA & D3PUBLISHER 春季特卖`](https://keylol.com/t939892-1-1))

另外 radar target function 写法是已弃用还是 bug 了?
现在好像都会变成 string

### CODE REVIEW

**Comment by kennyfong19931** [comment]
补个本地运行的结果
[keylol.txt](https://github.com/DIYgod/RSSHub/files/15200955/keylol.txt)

File: lib/routes/keylol/index.ts [hunk 1]
```diff
@@ -1,36 +1,61 @@
...
+const threadIdRegex = /(\d+)-\d+-\d+/;
 
 export const route: Route = {
-    path: '*',
-    name: 'Unknown',
-    maintainers: [],
+    path: '/:path',
+    name: '论坛',
+    parameters: { path: '路径，默认为热点聚焦' },
+    categories: ['game', 'bbs'],
```
Comment (TonyRL) [inline lib/routes/keylol/index.ts:16]:
```suggestion
    categories: ['game'],
```

File: lib/routes/keylol/index.ts [hunk 2]
```diff
@@ -44,12 +69,41 @@ async function handler(ctx) {
...
+                            );
+                        })
+                    );
+                } else {
+                    // normal post
+                    descriptionList.push(getDescription(content));
+                }
+
+                item.description = descriptionList.join('<br/>');
+                const authorName = authorNameMap.filter((a) => a.threadId === threadId);
```
Comment (TonyRL) [inline lib/routes/keylol/index.ts:105]: Please use `.find` instead

File: lib/routes/keylol/index.ts [hunk 3]
```diff
@@ -1,36 +1,61 @@
...
     const rootUrl = 'https://keylol.com';
-    const currentUrl = new URL(`forum.php?mod=forumdisplay&${thePath.replaceAll(/mod=\w+&/g, '')}`, rootUrl).href;
+    const currentUrl = queryString.stringifyUrl({ url: `${rootUrl}/forum.php`, query: queryParams });
 
     const { data: response } = await got(currentUrl);
 
     const $ = load(response);
 
-    let items = $('a.xst')
-        .slice(0, limit)
```
Comment (TonyRL) [inline lib/routes/keylol/index.ts:33]: Why remove the [commmon parameter limit](https://docs.rsshub.app/guide/parameters#limit-entries)?

File: lib/routes/keylol/index.ts [hunk 4]
```diff
@@ -44,12 +69,41 @@ async function handler(ctx) {
...
                 const content = load(detailResponse);
 
-                item.description = content('td.t_f').html();
-                item.author = content('a.xw1').first().text();
+                let descriptionList: any[] = [];
+                const indexDiv = content('div#threadindex');
+                if (indexDiv.length > 0) {
+                    // post with page
+                    const postId = content('div.t_fsz > script')
+                        .html()
```
Comment (TonyRL) [inline lib/routes/keylol/index.ts:82]:
```suggestion
                        .text()
```

File: lib/routes/keylol/index.ts (hunk 3 above)
Comment (kennyfong19931) [reply lib/routes/keylol/index.ts:33]: First thought it was useless as a limit is a common param handled by middleware.

Added back to reduce rest calls, especially for the post with a table of contents that I have added support to.

**Pull request merged**